components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.common/src/main/java/org/wso2/carbon/identity/api/server/action/management/common/factory/ActionMgtOSGiServiceFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.action.management.common.factory;
+
+import org.springframework.beans.factory.config.AbstractFactoryBean;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.action.mgt.ActionManager;
+
+/**
+ * Factory class for ActionManagementOSGiService.
+ */
+public class ActionMgtOSGiServiceFactory extends AbstractFactoryBean<ActionManager> {
+
+    private ActionManager actionManager;
+
+    @Override
+    public Class<?> getObjectType() {
+
+        return Object.class;
+    }
+
+    @Override
+    protected ActionManager createInstance() throws Exception {
+
+        if (this.actionManager == null) {
+            actionManager = (ActionManager) PrivilegedCarbonContext.
+                    getThreadLocalCarbonContext().getOSGiService(ActionManager.class, null);
+            if (actionManager == null) {
+                throw new Exception("Unable to retrieve ActionManager service.");
```

Comment:
Shall we have it as "ActionManager service not available"

Reply:
Fixed with https://github.com/wso2/identity-api-server/pull/626/commits/2dac506e1f7ffa6d587ddae09f7bf059c0f35f95